### ISSUE TITLE
ADR 102: RPC Companion architecture

### ISSUE THREAD

**Issue by thanethomson** [cometbft/cometbft#592]
As part of #574, we want to offer users a standalone proof-of-concept RPC service to serve as an example of a data companion. To this end, we want an ADR that captures:

1. The overall architecture of our standalone RPC service (implemented as a data companion)
2. Which of our [RPC endpoints](https://docs.cometbft.com/v0.34/rpc/) we'll be exposing as `/v1`
3. The decision as to which database to use to facilitate (1). The best database here will depend on the data access patterns defined by (1).

Future efforts could possibly involve productionizing this to the point that it may be able to scale independently of a full node, possibly reducing the costs associated with providing an RPC as a service, but this work needs to be broken down in separate follow-up issues.

**Comment by jmalicevic** [comment]
I am wondering whether the access patterns will depend on the endpoints exposed? If we have a preliminary list, I could make sure to gather the info on access patterns during the storage investigation.

**Comment by andynog** [comment]
I've started adding information about the RPC endpoints that will be implemented. I've used a table format because I believe it's easier to understand. There are a lot more info to add so it's WIP. For each endpoint I have to dig in into the code to understand it a bit better to assess the complexities on implement them. I also plan to add more sections outlining on how things will work, e.g. translating queries or return a response mapping the database records to a struct (or protobuf).
